src/main/java/fi/otavanopisto/kuntaapi/server/integrations/management/ManagementPageEntityUpdater.java

```diff
@@ -0,0 +1,176 @@
+import fi.otavanopisto.kuntaapi.server.system.SystemUtils;
+import fi.otavanopisto.mwp.client.ApiResponse;
+import fi.otavanopisto.mwp.client.DefaultApi;
+import fi.otavanopisto.mwp.client.model.Attachment;
+import fi.otavanopisto.mwp.client.model.Page;
+
+@ApplicationScoped
+@Singleton
+@AccessTimeout (unit = TimeUnit.HOURS, value = 1l)
+@SuppressWarnings ("squid:S3306")
+public class ManagementPageEntityUpdater extends EntityUpdater {
+
+  private static final int TIMER_INTERVAL = 1000;
+
+  @Inject
+  private Logger logger;
+  
+  @Inject
+  private ManagementApi managementApi;
+  
+  @Inject
+  private ManagementImageLoader managementImageLoader;
+  
+  @Inject
+  private IdentifierController identifierController;
+  
+  @Inject
+  private ModificationHashCache modificationHashCache;
+  
+  @Resource
+  private TimerService timerService;
+
+  private boolean stopped;
+  private List<PageIdUpdateRequest> queue;
+
+  @PostConstruct
+  public void init() {
+    queue = Collections.synchronizedList(new ArrayList<>());
+  }
+
+  @Override
+  public String getName() {
+    return "management-pages";
+  }
+
+  @Override
+  public void startTimer() {
+    startTimer(TIMER_INTERVAL);
+  }
+
+  private void startTimer(int duration) {
+    stopped = false;
+    TimerConfig timerConfig = new TimerConfig();
+    timerConfig.setPersistent(false);
+    timerService.createSingleActionTimer(duration, timerConfig);
+  }
+
+  @Override
+  public void stopTimer() {
+    stopped = true;
+  }
+  
+  @Asynchronous
+  public void onOrganizationIdUpdateRequest(@Observes PageIdUpdateRequest event) {
+    if (!stopped) {
+      PageId pageId = event.getId();
+      
+      if (!StringUtils.equals(pageId.getSource(), MwpConsts.IDENTIFIER_NAME)) {
+        return;
+      }
+      
+      if (event.isPriority()) {
+        queue.remove(event);
+        queue.add(0, event);
+      } else {
+        if (!queue.contains(event)) {
+          queue.add(event);
+        }
+      }
+    }
+  }
+  
+  @Timeout
+  public void timeout(Timer timer) {
+    if (!stopped) {
+      if (!queue.isEmpty()) {
+        PageIdUpdateRequest updateRequest = queue.remove(0);
+        updateManagementPage(updateRequest.getOrganizationId(), updateRequest.getId());
+      }
+
+      startTimer(SystemUtils.inTestMode() ? 1000 : TIMER_INTERVAL);
+    }
+  }
+  
+  private void updateManagementPage(OrganizationId organizationId, PageId pageId) {
+    DefaultApi api = managementApi.getApi(organizationId);
+    
+    ApiResponse<Page> response = api.wpV2PagesIdGet(pageId.getId(), null);
+    if (response.isOk()) {
+      updateManagementPage(api, response.getResponse());
+    } else {
+      logger.warning(String.format("Find organization %s page %s failed on [%d] %s", organizationId.getId(), pageId.toString(), response.getStatus(), response.getMessage()));
+    }
+  }
+  
+  private void updateManagementPage(DefaultApi api, Page managementPage) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "updateManagementPage" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)